packages/aws-cdk/lib/api/hotswap/cloudformation-executable-template.ts

```diff
@@ -0,0 +1,230 @@
+
+  private evaluateIntrinsic(intrinsic: Intrinsic): any {
+    const intrinsicFunc = (this as any)[intrinsic.name];
+    if (!intrinsicFunc) {
+      throw new Error(`CloudFormation function ${intrinsic.name} is not supported`);
+    }
+
+    const argsAsArray = Array.isArray(intrinsic.args) ? intrinsic.args : [intrinsic.args];
+
+    return intrinsicFunc.apply(this, argsAsArray);
+  }
+
+  private async findRefTarget(logicalId: string): Promise<string | undefined> {
+    // first, check to see if the Ref is a Parameter who's value we have
+    const parameterTarget = this.context[logicalId];
+    if (parameterTarget) {
+      return parameterTarget;
+    }
+    // if it's not a Parameter, we need to search in the current Stack resources
+    return this.findGetAttTarget(logicalId);
+  }
+
+  private async findGetAttTarget(logicalId: string, attribute?: string): Promise<string | undefined> {
+    const stackResources = await this.stackResources.listStackResources();
+    const foundResource = stackResources.find(sr => sr.LogicalResourceId === logicalId);
+    if (!foundResource) {
+      return undefined;
+    }
+    // now, we need to format the appropriate identifier depending on the resource type,
+    // and the requested attribute name
+    return this.formatResourceAttribute(foundResource, attribute);
+  }
+
+  private formatResourceAttribute(resource: AWS.CloudFormation.StackResourceSummary, attribute: string | undefined): string | undefined {
+    const physicalId = resource.PhysicalResourceId;
+
+    // no attribute means Ref expression, for which we use the physical ID directly
+    if (!attribute) {
+      return physicalId;
+    }
+
+    const resourceTypeFormats = RESOURCE_TYPE_ATTRIBUTES_FORMATS[resource.ResourceType];
+    if (!resourceTypeFormats) {
+      return physicalId;
+    }
+    const attributeFmtFunc = resourceTypeFormats[attribute];
+    if (!attributeFmtFunc) {
+      return physicalId;
+    }
+    const service = this.getServiceOfResource(resource);
+    const resourceTypeArnPart = this.getResourceTypeArnPartOfResource(resource);
+    return attributeFmtFunc({
+      partition: this.partition,
+      service,
+      region: this.region,
+      account: this.account,
+      resourceType: resourceTypeArnPart,
+      resourceName: physicalId!,
+    });
+  }
+
+  private getServiceOfResource(resource: AWS.CloudFormation.StackResourceSummary): string {
+    return resource.ResourceType.split('::')[1].toLowerCase();
+  }
+
+  private getResourceTypeArnPartOfResource(resource: AWS.CloudFormation.StackResourceSummary): string {
+    return resource.ResourceType.split('::')[2].toLowerCase();
+  }
+}
+
+interface ArnParts {
+  readonly partition: string;
+  readonly service: string;
+  readonly region: string;
+  readonly account: string;
+  readonly resourceType: string;
+  readonly resourceName: string;
+}
+
+const RESOURCE_TYPE_ATTRIBUTES_FORMATS: { [type: string]: { [attribute: string]: (parts: ArnParts) => string } } = {
```

Comment:
More tables to maintain ☹️ 

Unfortunately no way around it I suppose... but this is starting to feel dangerous, is it not?

What does SAM CLI do here?

Reply:
> Unfortunately no way around it I suppose... but this is starting to feel dangerous, is it not?

A little bit, but this should be fairly stable, so I'm not _that_ worried about it.

> What does SAM CLI do here?

I believe SAM CLI punts this completely to you, and forces you to evaluate these yourself.